pmd-core/src/main/java/net/sourceforge/pmd/RuleSetFactory.java

```diff
@@ -496,22 +508,46 @@ private void parseRuleSetReferenceNode(RuleSetReferenceId ruleSetReferenceId, Ru
         }
         final RuleSetReference ruleSetReference = new RuleSetReference(ref, true, excludedRulesCheck);
 
-        RuleSetFactory ruleSetFactory = new RuleSetFactory(this, warnDeprecated);
+        // load the ruleset with minimum priority low, so that we get all rules, to be able to exclude any rule
+        // minimum priority will be applied again, before constructing the final ruleset
+        RuleSetFactory ruleSetFactory = new RuleSetFactory(resourceLoader, RulePriority.LOW, warnDeprecated, this.compatibilityFilter != null);
```

Comment:
in the future, we should split the parsing of rules / references, from the addition to the current ruleset... possibly once we tackle #724